packages/eslint-plugin/src/rules/no-unnecessary-condition.ts

```diff
@@ -306,6 +326,8 @@ export default createRule<Options, MessageId>({
         messageId = !isUnaryNotArgument ? 'alwaysFalsy' : 'alwaysTruthy';
       } else if (!isPossiblyFalsy(type)) {
         messageId = !isUnaryNotArgument ? 'alwaysTruthy' : 'alwaysFalsy';
+      } else if (isFalseyBigInt(type)) {
```

Comment:
oh, fun. This line "_shouldn't_" be necessary, since for `0n`, `!isPossiblyTruthy(type)` _should_ be `true`, since ts-api-utils's `isFalsyType(type)` _should_ be `true`. But it's not, since ts-api-utils has the exact same bug that was in this rule!

https://github.com/JoshuaKGoldberg/ts-api-utils/blob/41cd21d41f445448681dd32308a48a004f4717cb/src/types/utilities.ts#L50

Filed https://github.com/JoshuaKGoldberg/ts-api-utils/issues/544 to get that corrected. I'd say - let's move this workaround to the `isPossiblyTruthy` function instead just after the `isFalsyType(type)` check, and put a note that this is a workaround for the ts-api-utils bug

Reply:
[Lines 75-77](https://github.com/typescript-eslint/typescript-eslint/pull/10205/files#diff-3cc43b506517aba406ae3a31d7998b116c4b09db76b632c6cb3c2e54fc71d66cR75-R77) implement the suggested change and note.